libobs/obs-bits.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#if defined(__GNUC__) || defined(__clang__)
+
+#define clz32(x) __builtin_clz(x)
+#define ctz32(x) __builtin_ctz(x)
+
+#elif defined(_MSC_VER) && _MSC_VER >= 1400
+
+static inline int clz32(unsigned long val)
+{
+	uint32_t zeros = 0;
+	_BitScanReverse(&zeros, val);
+	return 31 - zeros;
+}
+static inline int ctz32(unsigned long val)
+{
+	uint32_t zeros = 0;
+	_BitScanForward(&zeros, val);
+	return zeros;
+}
+
+#else
+
+static uint32_t popcnt(uint32_t x)
+{
+	x -= ((x >> 1) & 0x55555555);
+	x = (((x >> 2) & 0x33333333) + (x & 0x33333333));
+	x = (((x >> 4) + x) & 0x0f0f0f0f);
+	x += (x >> 8);
+	x += (x >> 16);
+	return x & 0x0000003f;
+}
+static uint32_t clz32(uint32_t x)
+{
+	x |= (x >> 1);
+	x |= (x >> 2);
+	x |= (x >> 4);
+	x |= (x >> 8);
+	x |= (x >> 16);
+	return 32 - popcnt(x);
+}
+static uint32_t ctz32(uint32_t x)
+{
+	return popcnt((x & -x) - 1);
+}
+
+#endif
+
+#ifndef MIN
+#define MIN(x, y) ((x) < (y) ? (x) : (y))
```

Comment:
Convert to a static inline function, please avoid macros and defines as much as possible, as there is no need for them in modern C/C++.

Reply:
Done in the last commit from yuriy-chumak